core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -908,19 +927,11 @@ public boolean hasPermission(Object issuer) {
     }
 
     public boolean hasPermission(CommandIssuer issuer) {
-        return permission == null || permission.isEmpty() || (manager.hasPermission(issuer, permission) && (parentCommand == null || parentCommand.hasPermission(issuer)));
+        return getRequiredPermissions().isEmpty() || getRequiredPermissions().stream().allMatch(permission -> manager.hasPermission(issuer, permission));
```

Comment:
I think this is premature optimization, calling `.stream()` on an empty Set will guarantee `allMatch` returns `true`>\.

This is known as a [vacuous truth](https://en.m.wikipedia.org/wiki/Vacuous_truth)

Reply:
< King of micro optimizations ;)

micro optimizations aren't something you "shouldn't" do, just more of a statement that it's hard to justify the time spent on doing it, or shouldn't do it if it makes the code harder to reason.

In this case, I asked him to do this to avoid the object allocation cost of .stream() and makes it clearer on intent without having to understand allMatch behavior for empty.